crates/ide_db/src/search.rs

```diff
@@ -252,23 +267,33 @@ impl<'a> FindUsages<'a> {
 
     pub fn at_least_one(self) -> bool {
         let mut found = false;
-        self.search(&mut |_reference| {
+        self.search(&mut |_, _| {
             found = true;
             true
         });
         found
     }
 
-    pub fn all(self) -> Vec<Reference> {
-        let mut res = Vec::new();
-        self.search(&mut |reference| {
-            res.push(reference);
+    /// The [`FileReferences`] returned always have unique [`FileId`]s.
+    pub fn all(self) -> Vec<FileReferences> {
+        let mut res = <Vec<FileReferences>>::new();
```

Comment:
o_O

That's a quite refined use of the turbofish! Let's stick to the more common `Vec::<FileReferences>::new();` though